app/lib/submission_builder/document.rb

```diff
@@ -114,6 +114,34 @@ def email_from_intake_or_df
       end
     end
 
+    def process_md_mailing_street(xml)
+      return unless @submission.data_source.direct_file_data.mailing_street.present?
+
+      mailing_street = sanitize_for_xml(@submission.data_source.direct_file_data.mailing_street)
+      key_position = mailing_street.index(ADDRESS_ABBREV_REGEX)
+
+      if key_position
+        truncated_mailing_street = mailing_street[0, key_position].rstrip
```

Comment:
[dust] minor style point, but for consistency with the below line (`[key_position..]`), can you use `[0...key_position]` here to denote the range between 0 inclusive and key_position non-inclusive? ([Range literal docs](https://ruby-doc.org/3.3.6/syntax/literals_rdoc.html#label-Range+Literals) because I always forget which one is which)